File: Algos/binarysearch.py
def binarysearch(arr,target):
    start = 0
    end = len(arr)-1
    while start <= end:
        mid = (end+start)//2
        if(target == arr[mid]):
            return mid
        if(arr[mid]>target):
            end = mid - 1
        else:
            start = mid + 1
    return -1
print(binarysearch([1,3,5,7,8,10],2))

File: Algos/BinarySearch.java
package Algos;

public class BinarySearch {
    public static int lower(int arr[],int key){
        int low = 0;
        int high = arr.length-1;
        while(low < high){
            int mid = low + (high - low)/2;
            if(arr[mid] >= key){
                high = mid;
            }
            else{
                low = mid+1;
            }
        }
        return low;
    }
}
